zng/any.go

```diff
@@ -0,0 +1,30 @@
+package zng
+
+import (
+	"errors"
+	"github.com/mccanne/zq/zcode"
+)
+
+var ErrInstantiateAny = errors.New("cannot instantiate type any")
+
+type TypeOfAny struct{}
+
+func (t *TypeOfAny) Parse(in []byte) (zcode.Bytes, error) {
+	return nil, ErrInstantiateAny
+}
+
+func (t *TypeOfAny) ID() int {
+	return IdAny
+}
+
+func (t *TypeOfAny) String() string {
+	return "any"
+}
+
+func (t *TypeOfAny) StringOf(zv zcode.Bytes) string {
+	return "-"
```

Comment:
This should decode the type and value from the zv and call the String method of said value and "-" if the value is not present.

Reply:
I thought we talked about starting with a baby step where TypeAny can only be used with the value null?  And that encoding some other type could come later?  (Its not clear to me how that would work exactly, we would need to touch a lot of procs and instances where type ids get mapped would need to either rewrite any values or have some way to propagate the original context to locations that need to read those.  That's all to say, it feels like a big task to me and out of scope for this patch)